lib/route_downcaser/downcase_route_middleware.rb

```diff
@@ -56,12 +55,16 @@ def downcased_uri(uri)
       return nil unless uri.present?
 
       if has_querystring?(uri)
-        "#{path(uri).mb_chars.downcase}?#{querystring(uri)}"
+        "#{downcased_path(uri)}?#{querystring(uri)}"
       else
-        path(uri).mb_chars.downcase
+        downcased_path(uri)
       end
     end
 
+    def downcased_path(uri)
+      path(uri).split("/").map{|segment| URI.encode_www_form_component(URI.decode_www_form_component(segment).downcase)}.join("/")
```

Comment:
Why do we need to split the path here? Is that so `"/"` doesn't get encoded?

Also there's a lot of splitting and joining and encoded and decoding happening here? Any chance we could make it a bit more obvious what's happening, perhaps extract a `downcase_path_segment` that does the actual downcasing:

    URI.encode_www_form_component(URI.decode_www_form_component(segment).downcase

?

Reply:
> Why do we need to split the path here? Is that so `"/"` doesn't get encoded?

Yes exactly. 

```ruby
URI.encode_www_form_component("one/two")
 => "one%2Ftwo"
```
 
> Also there's a lot of splitting and joining and encoded and decoding happening here? Any chance we could make it a bit more obvious what's happening, perhaps extract a `downcase_path_segment` that does the actual downcasing:
> 
> ```
> URI.encode_www_form_component(URI.decode_www_form_component(segment).downcase
> ```
> 
> ?

sure thing, will adjust and push.

PS what did you think of #45 - while this PR is doing a lot more processing than before (potential performance hit), removing the unnecessary transform of REQUEST_URI would then help to overset this (about 1/2 the processing)

Reply:
I think it should be possible to stop processing `REQUEST_URI`, as you're suggesting.